Mathlib/CategoryTheory/Types.lean

```diff
@@ -0,0 +1,405 @@
+can be viewed as a `LargeCategory` in our framework.
+
+Lean can not transparently view a function as a morphism in this category, and needs a hint in
+order to be able to type check. We provide the abbreviation `asHom f` to guide type checking,
+as well as a corresponding notation `↾ f`. (Entered as `\upr `.) 
+
+We provide various simplification lemmas for functors and natural transformations valued in `Type`.
+
+We define `uliftFunctor`, from `Type u` to `Type (max u v)`, and show that it is fully faithful
+(but not, of course, essentially surjective).
+
+We prove some basic facts about the category `Type`:
+*  epimorphisms are surjections and monomorphisms are injections,
+* `Iso` is both `Iso` and `Equiv` to `Equiv` (at least within a fixed universe),
+* every type level `IsLawfulFunctor` gives a categorical functor `Type ⥤ Type`
+  (the corresponding fact about monads is in `Mathlib/CategoryTheory/Monad/Types.lean`).
+-/
+
+
+namespace CategoryTheory
+
+-- morphism levels before object levels. See note [category_theory universes].
+universe v v' w u u'
+
+/- The `@[to_additive]` attribute is just a hint that expressions involving this instance can
+  still be additivized. -/
+@[to_additive CategoryTheory.types]
+instance types : LargeCategory (Type u)
+    where
+  Hom a b := a → b
+  id a := id
+  comp f g := g ∘ f
+#align category_theory.types CategoryTheory.types
+
+theorem types_hom {α β : Type u} : (α ⟶ β) = (α → β) :=
+  rfl
+#align category_theory.types_hom CategoryTheory.types_hom
+
+theorem types_id (X : Type u) : 𝟙 X = id :=
+  rfl
+#align category_theory.types_id CategoryTheory.types_id
+
+theorem types_comp {X Y Z : Type u} (f : X ⟶ Y) (g : Y ⟶ Z) : f ≫ g = g ∘ f :=
+  rfl
+#align category_theory.types_comp CategoryTheory.types_comp
+
+@[simp]
+theorem types_id_apply (X : Type u) (x : X) : (𝟙 X : X → X) x = x :=
+  rfl
+#align category_theory.types_id_apply CategoryTheory.types_id_apply
+
+@[simp]
+theorem types_comp_apply {X Y Z : Type u} (f : X ⟶ Y) (g : Y ⟶ Z) (x : X) : (f ≫ g) x = g (f x) :=
+  rfl
+#align category_theory.types_comp_apply CategoryTheory.types_comp_apply
+
+@[simp]
+theorem hom_inv_id_apply {X Y : Type u} (f : X ≅ Y) (x : X) : f.inv (f.hom x) = x :=
+  congr_fun f.hom_inv_id x
+#align category_theory.hom_inv_id_apply CategoryTheory.hom_inv_id_apply
+
+@[simp]
+theorem inv_hom_id_apply {X Y : Type u} (f : X ≅ Y) (y : Y) : f.hom (f.inv y) = y :=
+  congr_fun f.inv_hom_id y
+#align category_theory.inv_hom_id_apply CategoryTheory.inv_hom_id_apply
+
+-- Unfortunately without this wrapper we can't use `CategoryTheory` idioms, such as `IsIso f`.
+/-- `asHom f` helps Lean type check a function as a morphism in the category `Type`. -/
+abbrev asHom {α β : Type u} (f : α → β) : α ⟶ β :=
+  f
+#align category_theory.as_hom CategoryTheory.asHom
+
+-- If you don't mind some notation you can use fewer keystrokes:
+/-- Type as \upr in VScode -/
```

Comment:
VScode automatically gives suggestions about how to type unicode symbols as part of hover tooltips. So let's copy the docstring from above
```suggestion
@[inherit_doc]
```